### DISCUSSION TITLE
我如何让局域网指定IP之访问国内网络，不走代理

### DISCUSSION THREAD

**Discussion by gukey** [kkkgo/PaoPaoGateWay#67] [配置分享]
如题，看了下没找到设置；

**Comment by kkkgo** [comment]
控制局域网设备走不走代理并非是DNS的功能，如果你和PPGW配合使用而又想部分设备不获取到FAKEIP的解析结果的话，你为该设备设置另外的DNS即可，比如开启另一个DNS容器，在目标设备设备手动设置DNS，或者由路由器DHCP设置指定设备分配指定的DNS，具体设置需要看你路由器如何设置。

**Reply by kkkgo** [reply]
《Routeros-Dhcp-Server给个别用户定义不同的网关和DNS》
这篇文章也许是你需要的：https://www.roszj.com/1894.html

**Reply by gukey** [reply]
好的，我试试

**Comment by kkkgo** [comment]
openwrt用户可以参考：https://openwrt.org/docs/guide-user/base-system/dhcp_configuration#classifying_clients_and_assigning_individual_options1

**Reply by kkkgo** [reply]
```shell
#Client classifying and individual options
#Use the tag classifier to create a tagged group. 
#Assign individual DHCP options to hosts tagged with tag1. Specify custom DNS and possibly other DHCP options.

uci set dhcp.tag1="tag"
uci set dhcp.tag1.dhcp_option="6,8.8.8.8,8.8.4.4"
uci add dhcp host
uci set dhcp.@host[-1].name="j400"
uci set dhcp.@host[-1].mac="00:21:63:75:aa:17"
uci set dhcp.@host[-1].ip="10.11.12.14"
uci set dhcp.@host[-1].tag="tag1"
uci add dhcp host
uci set dhcp.@host[-1].name="j500"
uci set dhcp.@host[-1].mac="01:22:64:76:bb:18"
uci set dhcp.@host[-1].ip="10.11.12.15"
uci set dhcp.@host[-1].tag="tag1"
uci commit dhcp
/etc/init.d/dnsmasq restart
```

**Comment by kkkgo** [comment]
其他路由器用户： 如果主路由没有相关的设置项，可以考虑把主路由的DHCP服务禁用，使用第三方的DHCP服务来完成此功能。
比如可以运行Adguard home的docker，在设置-DHCP设置里可以开启DHCP服务，  
在设置-客户端设置里可以为不同客户端分配不同的DNS。

**Reply by mouyase** [reply]
如果主路由不支持添加静态路由，只用PaoPaoDNS能做到分流吗

**Reply by kkkgo** [reply]
静态路由似乎和这个贴子主题内容无关。
参考 https://github.com/kkkgo/PaoPaoGateWay/discussions/25
